__test__/api/comments/updateSingleCommentDownVote.test.js

```diff
@@ -0,0 +1,165 @@
+
+    // Delete cache.
+    oldComment = null;
+  });
+
+  it("Should add or remove a downvote to a comment (toggle)", async () => {
+    const url = `/v1/comments/${oldComment.commentId}/votes/downvote`;
+    const bearerToken = `bearer ${global.appToken}`;
+    const ownerIdUpdate = "voter@gmail.com";
+
+    // Run test matchers to verify that the comment votes hav not been updated in the database.
+    await CommentModel.findById(oldComment.commentId).then((comment) => {
+      expect(comment).toBeTruthy();
+      expect(comment.upVotes.length).toEqual(oldComment.numOfUpVotes);
+      expect(comment.downVotes.length).toEqual(oldComment.numOfDownVotes);
+      expect(comment.downVotes.length + comment.upVotes.length).toEqual(
+        oldComment.numOfVotes
+      );
+    });
+
+    // Run test matchers to verify that the comment votes addition produced the correct success response.
+    const addRes = await request
+      .patch(url)
+      .set("Authorization", bearerToken)
+      .send({
+        ownerId: ownerIdUpdate,
+      });
+
+    expect(addRes.status).toEqual(200);
+    expect(addRes.body.status).toEqual("success");
+    expect(addRes.body.data).toEqual({
+      commentId: oldComment.commentId,
+      numOfVotes: oldComment.numOfVotes + 1,
+      numOfUpVotes: oldComment.numOfUpVotes,
+      numOfDownVotes: oldComment.numOfDownVotes + 1,
+    });
+
+    // Run test matchers to verify that the comment votes have been added in the database.
+    await CommentModel.findById(oldComment.commentId).then((comment) => {
+      expect(comment).toBeTruthy();
+      expect(comment.upVotes.length).toEqual(oldComment.numOfUpVotes);
+      expect(comment.downVotes.length).toEqual(oldComment.numOfDownVotes + 1);
+      expect(comment.downVotes.length + comment.upVotes.length).toEqual(
+        oldComment.numOfVotes + 1
+      );
+    });
+
+    // Run test matchers to verify that the comment votes removal produced the correct success response.
+    const removeRes = await request
+      .patch(url)
+      .set("Authorization", bearerToken)
+      .send({
+        ownerId: ownerIdUpdate,
+      });
+
+    expect(removeRes.status).toEqual(200);
+    expect(removeRes.body.status).toEqual("success");
+    expect(removeRes.body.data).toEqual({
+      commentId: oldComment.commentId,
+      numOfVotes: oldComment.numOfVotes,
+      numOfUpVotes: oldComment.numOfUpVotes,
+      numOfDownVotes: oldComment.numOfDownVotes,
+    });
+
+    // Run test matchers to verify that the comment vote has been removed in the database.
+    await CommentModel.findById(oldComment.commentId).then((comment) => {
+      expect(comment).toBeTruthy();
+      expect(comment.upVotes.length).toEqual(oldComment.numOfUpVotes);
+      expect(comment.downVotes.length).toEqual(oldComment.numOfDownVotes);
+      expect(comment.downVotes.length + comment.upVotes.length).toEqual(
+        oldComment.numOfVotes
+      );
+    });
+  });
+
```

Comment:
I think one more test may be needed here for a scenario where the user has upvoted before and they are downvoting now. Then the `numOfVotes` will be unchanged and `numOfUpvotes` will be less by 1

Reply:
Yes! Nice catch.